### PR TITLE
New PrefabSpawnerComponent

### PR DESCRIPTION
**Pull request by VladimirLagutin**
## What does this PR do?

Implement new PrefabSpawnerComponent to attach it to any Entity in the Editor. The component spawns given prefab.
In LY it was SpawnerComponent which spawns dynamicslice.

## How was this PR tested?

In the Editor the component is added into the new Entity. The building prefab is set as argument and "Spawn on activate" and "Destroy on deactivate" are set to True. 
The "Play Game" button is pressed in the Editor. 
The building appears in the 3D world.
After pressing the Stop button the building is destroyed. No any Asserts appear in the log.

### CODE REVIEW

File: Code/Framework/AzFramework/AzFramework/Spawnable/Spawnable.h [hunk 1]
```diff
@@ -181,6 +181,17 @@ namespace AzFramework
         {
             m_isDynamic = value;
         }
+
+        static const char* GetFileFilter()
```
Comment (Aleks-Starykh-TR) [inline Code/Framework/AzFramework/AzFramework/Spawnable/Spawnable.h:186]: patch comments ?

Comment (VladimirLagutin) [reply Code/Framework/AzFramework/AzFramework/Spawnable/Spawnable.h:186]: Added

File: Gems/LmbrCentral/Code/Source/Scripting/PrefabSpawnerComponent.h [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+#include <LmbrCentral/Scripting/PrefabSpawnerComponentBus.h>
+#include <AzFramework/Spawnable/Spawnable.h>
+#include <AzFramework/Spawnable/SpawnableEntitiesInterface.h>
+
+namespace LmbrCentral
+{
+    /**
+    * PrefabSpawnerComponent
+    *
+    * PrefabSpawnerComponent facilitates spawning of a design-time selected or run-time provided "*.dynamicprefab" at an entity's location with an optional offset.
```
Comment (Aleks-Starykh-TR) [inline Gems/LmbrCentral/Code/Source/Scripting/PrefabSpawnerComponent.h]: *.dynamicprefab -> *.spawnable

File: Gems/LmbrCentral/Code/include/LmbrCentral/Scripting/PrefabSpawnerComponentBus.h [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        virtual AzFramework::EntitySpawnTicket SpawnRelative(const AZ::Transform& relative) = 0;
+
+        //! Spawn the selected prefab at the specified world transform
+        virtual AzFramework::EntitySpawnTicket SpawnAbsolute(const AZ::Transform& world) = 0;
+
+        //! Spawn the provided prefab at the entity's location
+        virtual AzFramework::EntitySpawnTicket SpawnPrefab(const AZ::Data::Asset<AzFramework::Spawnable>& prefab) = 0;
+
+        //! Spawn the provided prefab at the entity's location with the provided relative offset
+        virtual AzFramework::EntitySpawnTicket SpawnPrefabRelative(const AZ::Data::Asset<AzFramework::Spawnable>& slice, const AZ::Transform& relative) = 0;
```
Comment (Aleks-Starykh-TR) [inline Gems/LmbrCentral/Code/include/LmbrCentral/Scripting/PrefabSpawnerComponentBus.h]: Maybe rename argument from slice to prefab? (here and elsewhere, including comments)

File: Gems/LmbrCentral/Code/Source/Scripting/PrefabSpawnerComponent.h (hunk 2 above)
Comment (VladimirLagutin) [reply Gems/LmbrCentral/Code/Source/Scripting/PrefabSpawnerComponent.h]: Fixed

File: Gems/LmbrCentral/Code/include/LmbrCentral/Scripting/PrefabSpawnerComponentBus.h (hunk 3 above)
Comment (VladimirLagutin) [reply Gems/LmbrCentral/Code/include/LmbrCentral/Scripting/PrefabSpawnerComponentBus.h]: Of course

File: Gems/LmbrCentral/Code/include/LmbrCentral/Scripting/PrefabSpawnerComponentBus.h [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#if defined(CARBONATED)
+
+#include <AzCore/Component/ComponentBus.h>
+#include <AzCore/Math/Transform.h>
+#include <AzCore/Slice/SliceAsset.h>
```
Comment (Aleks-Starykh-TR) [inline Gems/LmbrCentral/Code/include/LmbrCentral/Scripting/PrefabSpawnerComponentBus.h]: Is this needed?
I'd expect here a header for spawnable.
Have you tried to compile Non-Unity build?

Comment (VladimirLagutin) [reply Gems/LmbrCentral/Code/include/LmbrCentral/Scripting/PrefabSpawnerComponentBus.h]: It is not.

**Review by Aleks-Starykh-TR — APPROVED** [review]
LGTM

**Review by alexeykostin — APPROVED** [review]

**Pull request merged**